src/plugins/fs/ext.c

```diff
@@ -59,6 +59,75 @@ static guint32 fs_mode_util[BD_FS_MODE_LAST+1] = {
 
 #define UNUSED __attribute__((unused))
 
+static gint8 compute_percents(guint8 pass_cur, guint8 pass_total, gint val_cur, gint val_total)
+{
+    gint perc;
+    gint one_pass;
+    /* first get a percentage in the current pass/stage */
+    perc = (val_cur * 100) / val_total;
+
+    /* now map it to the total progress, splitting the stages equally */
+    one_pass = 100 / pass_total;
+    perc = ((pass_cur - 1) * one_pass) + (perc / pass_total);
+
+    return perc;
+}
+
+/**
+ * filter_line_fsck: (skip)
+ * Filter one line - decide what to do with it.
+ *
+ * Returns: Zero or positive number as a percentage, -1 if not a percentage, -2 on an error
+ */
+static gint8 filter_line_fsck(const gchar * line, guint8 total_stages, GError **error) {
+    static GRegex *regex;
+    static int reginit = 0;
```

Comment:
Please rename ``regex`` to something that suggests it's specific for this function. Also, either change ``reginit`` to ``gboolean`` or get rid of it altogether (use ``regex != NULL``).